### PR TITLE
Prevent users from deleting other users' articles

### PR DESCRIPTION
**Pull request by gmcabrita**
Without this change any logged in user can delete another user's articles.

### CODE REVIEW

File: app/controllers/api/v1/articles_controller.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def create
   end
 
   def destroy
-    article = Article.find(params[:id])
+    article = Article.find_by(id: params[:id], user: current_user)
```
Comment (maximevaillancourt) [inline app/controllers/api/v1/articles_controller.rb:25]: `article` could be `nil` if there is no article that matches the search criteria. In this case, we should return a 404:

```suggestion
    article = Article.find_by(id: params[:id], user: current_user)
    
    unless article
      return render json: { error: 'article not found' }, status: :not_found
    end
```

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -62,7 +62,7 @@ def create
   end
 
   def destroy
-    @article = Article.find(params[:id])
+    @article = Article.find_by(id: params[:id], user: current_user)
     if @article.destroy
```
Comment (maximevaillancourt) [inline app/controllers/articles_controller.rb:66]: Similar here: we need to handle the case where `@article` is `nil` when there is no article that matches the search criteria:

```suggestion
    if @article&.destroy
```

**Comment by gmcabrita** [comment]
I'll handle these and add some tests later today :+1:

**Comment by gmcabrita** [comment]
I noticed the tests for the `Api::V1` controllers were missing the `Api::V1` namespacing which was preventing the `Api::V1::ArticlesControllerTest` tests from runnning so I added the correct namespacing.

**Pull request merged**